### PR TITLE
add some boolean logic ops, tweak others

### PR DESCRIPTION
**Pull request by DeltaNedas**
- rename != from not to =/=, a lot of players mix it up with bitwise not
- add section comments
- add ||, && and ! equivalents under the same names
- move pow from bitwise stuff to arithmetic

### CODE REVIEW

File: core/src/mindustry/logic/LogicOp.java [hunk 1]
```diff
@@ -4,24 +4,32 @@
 import arc.util.*;
 
 public enum LogicOp{
+    //section arithmetic
```
Comment (summetdev) [inline core/src/mindustry/logic/LogicOp.java:7]: how about this?
```
//region arithmetic
...
//endregion
```

File: core/src/mindustry/logic/LogicOp.java [hunk 2]
```diff
@@ -4,24 +4,32 @@
...
     sub("-", (a, b) -> a - b),
     mul("*", (a, b) -> a * b),
     div("/", (a, b) -> a / b),
     idiv("//", (a, b) -> Math.floor(a / b)),
     mod("%", (a, b) -> a % b),
+    pow("^", Math::pow),
+    //section boolean logic
     equal("==", (a, b) -> Math.abs(a - b) < 0.000001 ? 1 : 0, (a, b) -> Structs.eq(a, b) ? 1 : 0),
-    notEqual("not", (a, b) -> Math.abs(a - b) < 0.000001 ? 0 : 1, (a, b) -> !Structs.eq(a, b) ? 1 : 0),
+    notEqual("=/=", (a, b) -> Math.abs(a - b) < 0.000001 ? 0 : 1, (a, b) -> !Structs.eq(a, b) ? 1 : 0),
```
Comment (summetdev) [inline core/src/mindustry/logic/LogicOp.java:17]: maybe `!=`? since equal is `==` so it makes sense.

File: core/src/mindustry/logic/LogicOp.java [hunk 3]
```diff
@@ -4,24 +4,32 @@
...
     lessThan("<", (a, b) -> a < b ? 1 : 0),
     lessThanEq("<=", (a, b) -> a <= b ? 1 : 0),
     greaterThan(">", (a, b) -> a > b ? 1 : 0),
     greaterThanEq(">=", (a, b) -> a >= b ? 1 : 0),
-    pow("^", Math::pow),
+    lnot("!", (a, b) -> a < 0.000001 ? 1 : 0, (a, b) -> a == null ? 1 : 0),
+    lor("||", (a, b) -> a < 0.000001 || b < 0.000001 ? 1 : 0, (a, b) -> a != null || b != null),
```
Comment (Anuken) [inline core/src/mindustry/logic/LogicOp.java]: Why is this needed?

File: core/src/mindustry/logic/LogicOp.java (hunk 2 above)
Comment (Anuken) [inline core/src/mindustry/logic/LogicOp.java:17]: This is not clear at all.

Comment (DeltaNedas) [reply core/src/mindustry/logic/LogicOp.java:17]: the mathematical symbol is = with a slash through it

File: core/src/mindustry/logic/LogicOp.java (hunk 3 above)
Comment (DeltaNedas) [reply core/src/mindustry/logic/LogicOp.java]: if (!cond || !ptr)
if (block || enemy)

**Comment by DeltaNedas** [comment]
changed =/= to != as requested

File: core/src/mindustry/logic/LogicOp.java (hunk 3 above)
Comment (Anuken) [reply core/src/mindustry/logic/LogicOp.java]: That's exactly what `or` already does.

Comment (DeltaNedas) [reply core/src/mindustry/logic/LogicOp.java]: technically correct, but can be neither 0 nor 1, e.g. 0 | 2 = 2

Comment (Anuken) [reply core/src/mindustry/logic/LogicOp.java]: Why is that a problem? `0 | 2 = 2 != 0` -> will be evaluated as `true` in any condition

File: core/src/mindustry/logic/LogicOp.java (hunk 1 above)
Comment (DeltaNedas) [reply core/src/mindustry/logic/LogicOp.java:7]: ok

**Comment by Anuken** [comment]
Decided to use `b-and` for *binary and* instead, as that's more clear than just using "and".
